JAVA1/src/test/SampleTest.java

```diff
@@ -0,0 +1,52 @@
+package test;
+import Solution.FileSearchInHomeDirectory;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SampleTest {
+
+    public FileSearchInHomeDirectory p;
+    @Before
+    public void init(){
+        p = new FileSearchInHomeDirectory();
+    }
+    @After
+    public void print(){
+        System.out.println(p.getmResultArray());
+    }
+    //Multiple tests as new object required for every search.
+    @Test
+    public void findFilesTest1(){
+        assertTrue("Check file path",p.findFiles(".*\\.java","/home"));
+    }
+    @Test
+    public void findFilesTest2(){
```

Comment:
you can write all possible test cases for single method or functionality in single method.